handlers/k8s.go

```diff
@@ -0,0 +1,47 @@
+package handlers
+
+import (
+	"errors"
+	"time"
+
+	"github.com/deis/k8s-claimer/k8s"
+	"github.com/deis/k8s-claimer/leases"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+var (
+	errNoExpiredLeases = errors.New("no expired leases exist")
+)
+
+// findExpiredLease searches in the leases in the svc annotations and returns the cluster name of
```

Comment:
@jackfrancis this func is used to find an expired lease for a client to use. See (3) under the [leasing algorithm](https://github.com/deis/k8s-claimer/blob/master/doc/architecture.md#leasing-algorithm) section.

Also, what's the benefit to prioritizing stale leases over recently expired ones?

Finally, I think you brought up a bigger point on the order of cluster search. If we look for expired leases before clusters not in annotations, we're going to more aggressively recycle leases (even if we look for stale leases first). Now, assuming most leases get properly released, we favor a more fair recycling policy for clusters. I'm not sure it matters, but thought I'd bring it up as a point for discussion. I'm guessing we'll have to test both out in the wild. @helgi do you have any input here, based on previous experience?